src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const key of action['keysToRemove']) {
```

Comment:
You don't need to use square brackets for accessing static key
```suggestion
        for (const key of action.keysToRemove) {
```